crates/red_knot_python_semantic/src/types/property_tests.rs

```diff
@@ -305,6 +310,14 @@ mod stable {
         never_subtype_of_every_fully_static_type, db,
         forall types t. t.is_fully_static(db) => Type::Never.is_subtype_of(db, t)
     );
+
+    // For fully static types, they should be subtype of their union.
+    type_property_test!(
+        fully_static_types_are_subtype_of_their_union, db,
+        forall types s, t.
+            s.is_fully_static(db) & t.is_fully_static(db)
+            => s.is_subtype_of(db, union(db, s, t)) & t.is_subtype_of(db, union(db, s, t))
```

Comment:
Let's use boolean and (`&&`) rather than bitwise and `&` :-)

```suggestion
            s.is_fully_static(db) && t.is_fully_static(db)
            => s.is_subtype_of(db, union(db, s, t)) & t.is_subtype_of(db, union(db, s, t))
```